src/Microsoft.IdentityModel.Tokens/BaseConfigurationManager.cs

```diff
@@ -76,6 +111,24 @@ public BaseConfiguration LastKnownGoodConfiguration
             {
                 _lastKnownGoodConfiguration = value ?? throw LogHelper.LogArgumentNullException(nameof(value));
                 _lastKnownGoodConfigFirstUse = DateTime.UtcNow;
+
+                if (_lkgConfigurationCache == null)
+                    _lkgConfigurationCache = new EventBasedLRUCache<BaseConfiguration, DateTime>(LastKnownGoodConfigurationSizeLimit, TaskCreationOptions.None, BaseConfigurationComparer, true, maintainLRU: true);
```

Comment:
Adopt the EventBaseLRUCache here. @brentschmaltz @dannybtsai 
What is a reasonable value for default cache size limit?

Reply:
10 - 20

Reply:
I don't think we should set maintainLRU to true.

Reply:
Use defalut value(false) for maintainLRU